medusa/helpers/__init__.py

```diff
@@ -416,7 +416,7 @@ def make_dirs(path):
             try:
                 logger.debug(u"Folder {path} didn't exist, creating it", path=path)
                 os.makedirs(path)
-            except (OSError, IOError) as e:
+            except (OSError, IOError, WindowsError) as e:
                 logger.error(u"Failed creating {path} : {error!r}", path=path, error=e)
```

Comment:
This way we'll always log as error and it will be submitable by the user. That's the place that it should delegate to exception handler. All these catches should delegate to exception handler